### PR TITLE
Fix #664: Solved the UX issue with switch profile of navigation drawer

### LINKED ISSUE #664: UX issues with switch profile of navigation drawer
If we click on the switch profile button present in the navigation drawer and then close the dialog by clicking somewhere else on the screen, the dialog box gets dismissed, but the navigation drawer still remains open, with the switch profile option highlighted.

In this, if the dialog button is canceled or the dialog disappears by clicking somewhere else, then the last selection option should get re-selected. 
Example:
If you are in HomeFragment and click on `Switch Profile` in NavigationDrawer then the dialog disappears (like after clicking Cancel), in this case, HomeFragment should get re-selected. Similarly for other cases too. 

Screenshot_1581622192

### PR DESCRIPTION
**Pull request by abhinavraj23**
## Explanation
Fixes #664 - Solved the UX issue with switch profile of navigation drawer

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by rt4914 — CHANGES_REQUESTED** [review]
As suspected the issue was because of CircleCi. Please add test case for this new functionality in `NavigationDrawerTestActivityTest`

**Comment by abhinavraj23** [comment]
@rt4914 I have written the test cases as well, now you can review it

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt [hunk 1]
```diff
@@ -210,6 +210,17 @@ class NavigationDrawerTestActivityTest {
...
   }
 
+  @Test
+  fun testNavigationDrawerTestActivity_openNavigationDrawer_selectSwitchProfileMenu_showsExitToProfileChooserDialog_clickCancel_checkDrawerIsClosed(){
+    launch(NavigationDrawerTestActivity::class.java).use {
+      onView(withId(R.id.home_activity_drawer_layout)).perform(open())
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_message)).check(matches(isDisplayed()))
+      onView(withText(R.string.home_activity_back_dialog_cancel)).perform(click())
+      onView(withId(R.id.home_activity_drawer_layout)).check(matches(isClosed()))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:221]: You will need to add one more line to check that `Home` is highlighted when you open drawer again. That was one of the main issues that we are fixing here. Also, you will need to update your test case name accordingly

**Review by rt4914 — CHANGES_REQUESTED** [review]
LGTM, one change needed in test case

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt (hunk 1 above)
Comment (abhinavraj23) [reply app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:221]: Changes done @rt4914

**Review by veena14cs — APPROVED** [review]
Code looks good to me. But Circleci is failing still. It is the same test which was failing earlier.

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt (hunk 1 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:221]: You are checking the content on home fragment, that is incorrect. We need to check that the navigation drawer has "Home" as highlighted.

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL

**Comment by veena14cs** [comment]
@rt4914 it has to close drawer on click on cancel on dialog  or anywhere outside right?

**Comment by rt4914** [comment]
@abhinavraj23 take latest code from develop and merge in this branch.

**Comment by abhinavraj23** [comment]
@rt4914 I have made the changes in the tests and have even updated it with the latest code from develop

File: app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt (hunk 1 above)
Comment (abhinavraj23) [reply app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:221]: @rt4914 I have made the change

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/app/testing/NavigationDrawerTestActivityTest.kt:221]: Actually this is better, but I had something else in mind. But i believe we cannot write the test case that I am thinking until we have Screenshot testing.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**